agent/consul/multilimiter/multilimiter.go

```diff
@@ -88,13 +100,35 @@ func NewMultiLimiter(c Config) *MultiLimiter {
 	if c.ReconcileCheckInterval == 0 {
 		c.ReconcileCheckLimit = 1 * time.Second
 	}
-	m := &MultiLimiter{limiters: &limiters, defaultConfig: &config, limitersConfigs: &configs}
+	chLimiter := make(chan *limiterWithKey, 100)
+	m := &MultiLimiter{limiters: &limiters, defaultConfig: &config, limitersConfigs: &configs, limiterCh: chLimiter, configsLock: sync.Mutex{}}
+
 	return m
 }
 
 // Run the cleanup routine to remove old entries of Limiters based on ReconcileCheckLimit and ReconcileCheckInterval.
 func (m *MultiLimiter) Run(ctx context.Context) {
+	isRunning := m.isRunning.Load()
+	if isRunning {
+		return
+	}
+	m.isRunning.Store(true)
 	go func() {
+		defer m.isRunning.Store(false)
```

Comment:
We do this in _both_ goroutines, which seems like it could lead to some unexpected behavior. I'd argue we probably don't need to re-run once stopped, but if we do we should use something like a `sync.WaitGroup` to wait for both goroutines to exit.